region/maridia/inner-green/East Sand Hall.json

```diff
@@ -814,6 +870,80 @@
       ],
       "devNote": "It is assumed that getting to the second pillar is free relative to getting to the right door."
     },
+    {
+      "link": [1, 4],
+      "name": "Cross Room Space Jump",
+      "entranceCondition": {
+        "comeInSpaceJumping": {
+          "speedBooster": false,
+          "minTiles": 6.4375
+        }
+      },
+      "requires": [
+        "canPreciseSpaceJump",
+        "canInsaneJump",
+        "canTrickyDodgeEnemies"
+      ],
+      "flashSuitChecked": true,
+      "note": [
+        "Space Jump into the room with full run speed without Speed Booster,",
+        "jumping as low as possible through the doorway.",
+        "Aim down when close to bonking the ceiling.",
+        "Samus should land on the first pillar."
+      ]
+    },
+    {
+      "link": [1, 4],
+      "name": "Cross Room Space Jump into Ceiling Mockball",
+      "entranceCondition": {
+        "comeInSpaceJumping": {
+          "speedBooster": false,
+          "minTiles": 5.4375
+        }
+      },
+      "requires": [
+        "canPreciseSpaceJump",
+        "canMomentumConservingMorph",
+        "canTrickyDodgeEnemies",
+        "canPlayInSand"
+      ],
+      "flashSuitChecked": true,
+      "note": [
+        "Space Jump into the room relatively low through the doorway.",
+        "Do a momentum-conserving morph on the ceiling."
```

Comment:
This may make people overthink the morph.  You are aiming for the slopey ceiling trap to morph against instead of the flat ceiling, so its a regular morph as soon as you enter the room.

Reply:
Added some more detail to the note to try to clarify. From what I can tell, morphing too early will bonk the corner and fail, so it does need to be a momentum conserving morph, and you're right it's about sliding past the corner rather than the sandfall ceiling. https://videos.maprando.com/video/8138